src/agent.rs

```diff
@@ -165,3 +179,38 @@ impl Agent {
         output.clone()
     }
 }
+
+fn check_cipher_blacklist(cipher: &str, shim: &str) -> bool {
+    let mut bl = fs::File::open("cipher_blacklist.json").unwrap();
+    let mut bls = String::from("");
+    bl.read_to_string(&mut bls)
+        .expect("Could not read file to string");
+    let blacklist: CipherBlacklist = json::decode(&bls).expect("Malformed JSON blacklist file.");
+    
+    if shim.contains("bssl_shim") {
+        for c in blacklist.bssl_blacklist.unwrap_or_else(|| {
```

Comment:
Not sure how the json parser is implemented, but we could avoid all these loops if you'd use a different json format. 

    {
      "ciphersuites": {
        "TLS_DHE_DSS_WITH_AES_128_GCM_SHA256": [
          "nss", "bssl", "ossl"
        ]
      }
    }

Reply:
Implemented this new format. It also allows comments for single ciphers, making it easier to investigate why certain ciphers are currently failing.